src/Agent.php

```diff
@@ -237,8 +242,31 @@ public function performAction($action_id, Map $arguments = null)
                 return $this->getActuator($actuator)->perform($action_id, $arguments);
             }
         }
+
+        $this->logger->info(sprintf('No supporting Store found for information request %', $action_id));
+        return null;
+    }
+
+    /**
+     * Loops through all available information requests and if they support the requested
+     *
+     * @param $information_request_id
+     * @param Map $utterance
+     * @return mixed|null
+     */
+    public function performInformationRequest($information_request_id, Map $utterance)
```

Comment:
An information request might come from something other than an utterance. Still sensible to standardise on a Map data type but don't need to call it utterance here - maybe just arguments?